ronal/core.py

```
@@ -29,6 +29,47 @@
 import logging
 import os
 import shutil
+import requests
+
+
+def call_navitia(url, auth=()):
+    try:
+        response = requests.get(url, auth=auth)
+    except requests.exceptions.ConnectionError:
+        logging.exception('error connecting: url {}'.format(url))
+        return
+    except requests.RequestException:
+        logging.exception('error fetching from navitia the last dataset production period')
+        return
+    else:
+        if response.status_code == 200:
+            return response
+        else:
+            logging.error('error fetching from navitia the last dataset production period: status code {}'.format(
+                response.status_code))
+            return
+
+
+def create_dir(directory):
+    """create directory if needed"""
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def _create_production_period(datasets):
+    dataset = next(iter(datasets.get('datasets', [])), None)
+    if dataset :
+        start_validation_date = dataset.get('start_validation_date')
+        end_validation_date = dataset.get('end_validation_date')
+        last_dataset_production_date = ProductionPeriod(start_validation_date, end_validation_date)
+        return last_dataset_production_date
+    return ProductionPeriod()
```

I would raise an error here, because we won't be able to do anything if we don't get a production period from navitia no ?